crates/core/src/network/handler/fetch.rs

```diff
@@ -2,30 +2,40 @@ use crate::{
     action::{Action, ActionWrapper},
     context::Context,
     instance::dispatch_action,
-    network::handler::{get_content_aspect, get_meta_aspects},
+    network::handler::{
+        get_content_aspect, get_meta_aspects_from_chain, get_meta_aspects_from_dht_eav,
+    },
 };
+use holochain_core_types::network::entry_aspect::EntryAspect;
 use lib3h_protocol::data_types::FetchEntryData;
-use std::sync::Arc;
+use std::{collections::HashSet, sync::Arc};
 
 /// The network has requested a DHT entry from us.
 /// Lets try to get it and trigger a response.
 pub fn handle_fetch_entry(get_dht_data: FetchEntryData, context: Arc<Context>) {
     let address = get_dht_data.entry_address.clone();
-    let mut aspects = vec![];
+    let mut aspects: HashSet<EntryAspect> = HashSet::new();
```

Comment:
Because of this we need the `Eq` derivations and the `Hash` implementation below.